File: tribonacci_sequence/tribonacci.hs
-- https://www.codewars.com/kata/556deca17c58da83c00002db
module Tribonacci where

fibs = map fst $ iterate (\(x,y,z) -> (y,z,x+y+z)) (a,b,c)

tribonacci :: Num a => (a, a, a) -> Int -> [a]
tribonacci (a, b, c) n = take n (fibs)

File: tribonacci_sequence/tribonacci.clj
(ns kata)

;; https://www.codewars.com/kata/556deca17c58da83c00002db

(defn generate_trib
  [a b c]
  (lazy-seq (cons a (generate_trib b c (+ a b c)))))

(defn tribonacci
  [[a b c] n]
  (take n (generate_trib a b c)))
